AppImage build failing

Last time it built successfully was 2 months ago. First failure was 2 months ago in https://github.com/netpanzer/netpanzer/actions/runs/10109062709

The build runs on Ubuntu Focal in this [linuxdeploy container](https://github.com/andy5995/linuxdeploy-build-helper-container). The NP build seems to be fine everywhere else.

```
[69/232] Compiling C++ object src/NetPanzer/libnetpanzer.so.p/.._Lib_2D_CachedFontRenderer.cpp.o
FAILED: src/NetPanzer/libnetpanzer.so.p/.._Lib_2D_CachedFontRenderer.cpp.o 
c++ -Isrc/NetPanzer/libnetpanzer.so.p -Isrc/NetPanzer -I../src/NetPanzer -I. -I.. -I../src/Lib -I/usr/include/lua5.1 -I/usr/include/SDL2 -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c++17 -O3 -fno-common -Wno-shadow -fPIC -D_REENTRANT -MD -MQ src/NetPanzer/libnetpanzer.so.p/.._Lib_2D_CachedFontRenderer.cpp.o -MF src/NetPanzer/libnetpanzer.so.p/.._Lib_2D_CachedFontRenderer.cpp.o.d -o src/NetPanzer/libnetpanzer.so.p/.._Lib_2D_CachedFontRenderer.cpp.o -c ../src/Lib/2D/CachedFontRenderer.cpp
../src/Lib/2D/CachedFontRenderer.cpp: In static member function ‘static SDL_Surface* CachedFontRenderer::render(const char*, SDL_Color, SDL_Color, bool, int)’:
../src/Lib/2D/CachedFontRenderer.cpp:81:13: error: ‘TTF_RenderUTF8_Shaded_Wrapped’ was not declared in this scope; did you mean ‘TTF_RenderUTF8_Blended_Wrapped’?
   81 |           ? TTF_RenderUTF8_Shaded_Wrapped(CachedFontRenderer::font, text, color, blendColor, wrapLength)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |             TTF_RenderUTF8_Blended_Wrapped
```

--- a/.github/workflows/appimage.yml
+++ b/.github/workflows/appimage.yml
@@ -17,6 +17,7 @@ on:
     paths:
     - '**/appimage.yml'
     - '**/AppRun'
+    - '**/make-appimage.sh'
 
 env:
   VERSION: ${{ inputs.version }}

--- a/.github/workflows/c-cpp.yml
+++ b/.github/workflows/c-cpp.yml
@@ -72,10 +72,17 @@ jobs:
         sudo apt update
         sudo apt upgrade -y
         sudo apt install -y \
-          meson \
-          ninja-build \
           python3 \
           python3-setuptools
+        if [ "${{ contains(matrix.build.name, '20.04') }}" = "false" ]; then
+          sudo apt install -y \
+          meson \
+          ninja-build
+        else
+          sudo apt install -y python3-pip
+          python -m pip install --upgrade pip
+          pip install meson ninja
+        fi
 
     - if: ${{ ! contains(matrix.build.name, 'no-deps') }}
       run: |

--- a/.github/workflows/c-cpp.yml
+++ b/.github/workflows/c-cpp.yml
@@ -29,6 +29,8 @@ jobs:
       fail-fast: true
       matrix:
         build:
+        - name: ubuntu-20.04-gcc
+          os: ubuntu-20.04  # supported until >= April 2025
         - name: data-install-test-gcc
           os: ubuntu-22.04
         - name: ubuntu-22.04-gcc

--- a/meson.build
+++ b/meson.build
@@ -97,13 +97,14 @@ if not phyfs_dep.found()
 endif
 deps += phyfs_dep
 
-sdl2_dep = dependency('SDL2', required: false)
+sdl2_dep = dependency('SDL2', version: '>=2.0.18', required: false)
 if not sdl2_dep.found()
   sdl2_dep = cc.find_library('SDL2', has_headers: 'SDL.h', required: false)
   if not sdl2_dep.found()
     # This will use the fallback
     sdl2_dep = dependency(
       'sdl2',
+      version: '>=2.0.18',
       required: true,
       fallback: 'sdl2',
       default_options: 'default_library=static')
@@ -128,7 +129,7 @@ if platform_is_windows
 endif
 # Don't add to deps, but add it to netpanzer_dep when the dependency is declared
 
-sdl2_ttf_dep = dependency('SDL2_ttf', required: false)
+sdl2_ttf_dep = dependency('SDL2_ttf', version: '>=2.1', required: false)
 if not sdl2_ttf_dep.found()
   sdl2_ttf_dep = cc.find_library('SDL2_ttf', has_headers: 'SDL_ttf.h', required: false)
   if not sdl2_ttf_dep.found()
@@ -154,9 +155,9 @@ if not sdl2_ttf_dep.found()
     endif
     deps += freetype2_dep
 
-    # Now use the sdl2_ttf fallback
     sdl2_ttf_dep = dependency(
       'sdl2_ttf',
+      version: '>=2.1',
       required: true,
       fallback: 'sdl2_ttf',
       default_options: ['default_library=static']

--- a/support/appimage/make-appimage.sh
+++ b/support/appimage/make-appimage.sh
@@ -42,14 +42,15 @@ else
 fi
 
 # Install necessary dependencies
+# Required version of SDL2 and and SDL2-ttf are too old in Ubuntu Focal, which is
+# the container distro for linuxdeploy.
+# See https://github.com/netpanzer/netpanzer/issues/235
 sudo apt update && sudo apt upgrade -y
 sudo apt install -y \
   gettext \
   liblua5.1-0-dev \
   libphysfs-dev \
-  libsdl2-dev \
-  libsdl2-mixer-dev \
-  libsdl2-ttf-dev
+  libsdl2-mixer-dev
 
 # Set up build directory
 BUILD_DIR="$SOURCE_ROOT/appimage_build"

